components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2276,13 +2310,31 @@ private Group getGroupWithName(String groupName)
         }
 
         Group group = new Group();
-        group.setDisplayName(groupName);
+        if (MandateDomainForUsernamesAndGroupNamesInResponse()) {
+            groupName = prependPrimaryDomain(groupName);
+            group.setDisplayName(groupName);
+        } else if (MandateDomainForGroupNamesInGroupsResponse()) {
+            group.setDisplayName(groupName);
+        } else {
+            if (StringUtils.isNotEmpty(userStoreDomainName) && (userStoreDomainName.toUpperCase()
+                    .equals(UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME)) && (
+                    groupName.indexOf(CarbonConstants.DOMAIN_SEPARATOR) > 0)) {
+                int indexOfDomainSeparator = groupName.indexOf(CarbonConstants.DOMAIN_SEPARATOR);
+                String role = groupName.substring(indexOfDomainSeparator + 1);
+                group.setDisplayName(role);
+            } else {
+                group.setDisplayName(groupName);
+            }
+        }
```

Comment:
From this block seems even for primary user store user store domain is prepended and it's being removed again here.
From the user core level usually primary is not appended for a domain. That means from the SCIM level itself is should have been added.
Rather being removing an added one better to avoid prepending the domain itself